force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +198,12 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Integer lowNum = 50;
+        Integer highNum = 100;
         // Return the random integer
-        return null; // Replace null with the variable you used to store the result
+        Integer randomNum = (Integer)Math.floor(Math.random() * (highNum - lowNum + 1)) + lowNum;
+        System.debug(randomNum);
+        return randomNum; // Replace null with the variable you used to store the result
```

Comment:
Good work. I go back and forth about declaring variables like lowNum inside a method body unless I think they are essential to reliability or the name gives you a lot of context about what that number represents.